ui/app/templates/storage/index.hbs

```diff
@@ -0,0 +1,233 @@
+        <h3>CSI Volumes</h3>
+        <p class="intro">
+          Storage configured by plugins run as Nomad jobs, with advanced features like snapshots and resizing.
+          <Hds::Link::Inline @href="https://developer.hashicorp.com/nomad/docs/concepts/plugins/csi" @icon="docs-link" @iconPosition="trailing">Read more</Hds::Link::Inline>
+        </p>
+        <div class="search">
+          <Hds::Form::TextInput::Field
+            data-test-csi-volumes-search
+            @type="search"
+            @value={{this.csiFilter}}
+            placeholder="Search CSI Volumes"
+            {{on "input" (action this.applyFilter "csi")}}
+          />
+        </div>
+      </header>
+      {{#if this.sortedCSIVolumes.length}}
+        <Hds::Table @caption="CSI Volumes"
+          @model={{this.paginatedCSIVolumes}}
+          @columns={{this.csiColumns}}
+          @sortBy={{this.csiSortProperty}}
+          @sortOrder={{if this.csiSortDescending "desc" "asc"}}
+          @onSort={{action this.handleSort "csi"}}
+          >
+          <:body as |B|>
+            <B.Tr data-test-csi-volume-row>
+              <B.Td data-test-csi-volume-name>
+                <LinkTo
+                  @route="storage.volumes.volume"
+                  @model={{B.data.idWithNamespace}}
+                  class="is-primary"
+                >
+                  {{B.data.plainId}}
+                </LinkTo>
+              </B.Td>
+              {{#if this.system.shouldShowNamespaces}}
+                <B.Td data-test-csi-volume-namespace>
+                  {{B.data.namespace.name}}
+                </B.Td>
+              {{/if}}
+              <B.Td data-test-csi-volume-schedulable>
+                {{if B.data.schedulable "Schedulable" "Unschedulable"}}
+              </B.Td>
+              <B.Td data-test-csi-volume-controller-health>
+                {{#if B.data.controllerRequired}}
+                  {{if (gt B.data.controllersHealthy 0) "Healthy" "Unhealthy"}}
+                  (
+                  {{B.data.controllersHealthy}}
+                  /
+                  {{B.data.controllersExpected}}
+                  )
+                {{else if (gt B.data.controllersExpected 0)}}
+                  {{if (gt B.data.controllersHealthy 0) "Healthy" "Unhealthy"}}
+                  (
+                  {{B.data.controllersHealthy}}
+                  /
+                  {{B.data.controllersExpected}}
+                  )
+                {{else}}
+                  <em class="is-faded">
+                    Node Only
+                  </em>
+                {{/if}}
+              </B.Td>
+              <B.Td data-test-csi-volume-node-health>
+                {{if (gt B.data.nodesHealthy 0) "Healthy" "Unhealthy"}}
+                (
+                {{B.data.nodesHealthy}}
+                /
+                {{B.data.nodesExpected}}
+                )
+              </B.Td>
+              <B.Td data-test-csi-volume-provider>
+                {{B.data.provider}}
```

Comment:
Is it feasible to link to the plugin here?

Reply:
Yeah; I'll still show .provider as the text, but link to the plugin by .pluginID I think

Reply:
Oh! I thought provider was the same as plugin and there was just a difference between CSI and DHV again. But now I'm remembering that it's not -- provider is a property of the CSI plugin that we copy over to the CSI volume. I suspect we should just show plugin ID in the table instead, as that's going to be more relevant to cluster admins than the provider. Then show provider in the detail page if you want.

Reply:
Ended up switching the title to Plugin and keeping the link, thanks for the suggestion!